test/pagination.ts

```diff
@@ -333,6 +333,74 @@ test('allowGetBody sends json payload with .paginate()', withBodyParsingServer,
 	t.deepEqual(results, [1, 2, 3]);
 });
 
+test.failing('`hooks` are not duplicated', withBodyParsingServer, async (t, server, got) => {
+	let page = 1;
+	server.get('/', (_request, response) => {
+		response.end(JSON.stringify([page++]));
+	});
+
+	const nopHook = () => {};
+
+	const result = await got.paginate.all<number>({
+		pagination: {
+			paginate: response => {
+				if ((response.body as string) === '[3]') {
+					return false; // Stop after page 3
+				}
+
+				const {options} = response.request;
+				const {init, beforeRequest, beforeRedirect, beforeRetry, afterResponse, beforeError} = options.hooks;
+				const hooksCount = [init, beforeRequest, beforeRedirect, beforeRetry, afterResponse, beforeError].map(a => a.length);
+
+				t.deepEqual(hooksCount, [1, 1, 1, 1, 1, 1]);
+
+				return options;
+			}
+		},
+		hooks: {init: [nopHook], beforeRequest: [nopHook], beforeRedirect: [nopHook], beforeRetry: [nopHook], afterResponse: [r => r], beforeError: [error => error]}
```

Comment:
This is an ugly line. Can you make this beautiful? Hint: use tabs :P

Reply:
I shall summon the Prettier thing...

https://www.youtube.com/watch?v=Q_AqEojDBjg